Sources/SwiftInfo/main.swift

```diff
@@ -1,24 +1,48 @@
+import ArgumentParser
 import Foundation
 import SwiftInfoCore
 
 let task = Process()
 
-public struct Main {
-    static func run() {
-        let fileUtils = FileUtils()
-        let toolchainPath = getToolchainPath()
-        log("SwiftInfo 2.3.12")
-        if ProcessInfo.processInfo.arguments.contains("-version") {
-            exit(0)
+struct SwiftInfo: ParsableCommand {
+
+    @Flag(name: .long, help: "return current version of `SwiftInfo`")
+    var version = false
+
+    @Flag(name: .shortAndLong, help: "silent all logs")
+    var silent = false
+
+    @Flag(name: .shortAndLong, help: "logs all details to console")
+    var verbose = false
+
+    @Flag(name: .long, help: "is in pull request mode")
+    var pullRequest = false
+
+    @Flag(name: .long, help: "print source kit")
+    var sourceKit = false
+
+    @Argument(help: "One or more user related Swiftc Args")
+    var arguments: [String] = []
```

Comment:
I don't understand why this is necessary, shouldn't we keep using `let processInfoArgs = ProcessInfo.processInfo.arguments` (or CommandLine.arguments) like before?

Reply:
Oh I thought that the main arguments are going to be only for `Swiftinfo` and we won't be passing them to the subprocess.